### ISSUE TITLE
Dropdown component should allow the role to be overriden (Accessibility)

### ISSUE THREAD

**Issue by tdroeske** [OfficeDev/office-ui-fabric-react#1268]
### Bug Report

- __Package version(s)__: 1.14

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)
No

Requested priority: (Blocking, High, Normal, Low) 
Normal

#### Describe the issue:

The Dropdown component has a role of "combobox". Our accessibility guidelines say that the role should be "listbox" since it is a simple dropdown and there is not editable aspect to it. We need to override the role but it isn't exposed.

#### Actual behavior:

Role is hard coded as "combobox".

#### Expected behavior:

Consumers should be able to override the role so accessibility needs can be met.

**Comment by mdahamiwal** [comment]
It is currently implemented as a readonly combobox which is the right pattern here. [`listbox`](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox) doesn't specify a option selector or droppable feature.

**Comment by hross** [comment]
@mdahamiwal I think we need to add the aria-readonly attribute to the dropdown:

http://www.ssbbartgroup.com/blog/differences-aria-1-0-1-1-changes-rolecombobox/

http://whatsock.com/tsg/Coding%20Arena/ARIA%20Comboboxes/ARIA%20Comboboxes%20(Native%20Inputs,%20Editable%20and%20Readonly)/demo.htm

to make it work as expected with the spec

**Comment by hross** [comment]
Another example with readonly set: http://www.oaa-accessibility.org/examplep/combobox6/
